### PR TITLE
Moved collector into subpackage with tests

### PR DESCRIPTION
**Pull request by lox**
Turns out not having tests sucks, as does having loads of code in the main package. This remedies both of these things to a minimal extent.

### CODE REVIEW

File: collector/collector.go [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+		r.Queues[queue][BusyAgentCount] = 0
+		r.Queues[queue][IdleAgentCount] = 0
+		r.Queues[queue][TotalAgentCount] = 0
+	}
+
+	err := p.Pages(func(v interface{}) bool {
+		agents := v.([]bk.Agent)
+
+		for _, agent := range agents {
+			queue := "default"
```
Comment (toolmantim) [inline collector/collector.go]: An agent can have multiple queues. I have this bug too in the new queues API endpoint, so the numbers were a little off. We'll need to fix that here too.

Comment (lox) [inline collector/collector.go]: ORLY? That's interesting. I will keep this behaviour in currently and look to fix it in a subsequent PR.

File: collector/collector.go [hunk 2]
```diff
@@ -0,0 +1,360 @@
...
+	}
+	agentService interface {
+		List(org string, opt *bk.AgentListOptions) ([]bk.Agent, *bk.Response, error)
+	}
+}
+
+func New(c *bk.Client, opts Opts) *Collector {
+	return &Collector{
+		Opts:         opts,
+		buildService: c.Builds,
```
Comment (callumj) [inline collector/collector.go]: Just noticed this in my local copy (I'm doing a fork to add statsd support): should `agentService` be populated too?

Comment (lox) [inline collector/collector.go]: Hrm, good point, now to figure out how it works at all.

Comment (lox) [inline collector/collector.go]: See https://github.com/buildkite/buildkite-metrics/pull/23 @callumj

**Pull request merged**